crypto/cipher_extra/tls_cbc.c

```diff
@@ -498,9 +498,168 @@ int EVP_tls_cbc_digest_record_sha256(
   return 1;
 }
 
+int EVP_final_with_secret_suffix_sha384(SHA512_CTX *ctx,
```

Comment:
@dkostic Could use some eyes on this function as we discussed offline, there seems to be some occasional failures in the CI on subsequent reruns around a bad computed MAC, which implies this function isn't 100% perfect.